includes/wc-cart-functions.php

```diff
@@ -481,3 +481,18 @@ function wc_shipping_methods_have_changed( $key, $package ) {
 	WC()->session->set( 'previous_shipping_methods', $previous_shipping_methods );
 	return $new_rates !== $prev_rates;
 }
+
+/**
+ * Gets a hash of important product data that when changed should cause cart items to be invalidated.
+ *
+ * The woocommerce_cart_item_data_to_validate filter can be used to add custom properties.
+ *
+ * @param WC_Product $product Product object.
+ * @return string
+ */
+function wc_get_cart_item_data_hash( $product ) {
+	return md5( wp_json_encode( apply_filters( 'woocommerce_cart_item_data_to_validate', array(
+		'price'      => $product->get_type(),
```

Comment:
I assume you meant to add a price and a type to the array here?

Reply:
I don't think price should remove it from cart.